### ISSUE TITLE
Error using sample function Richard.js

### ISSUE THREAD

**Issue by louisvangeldrop** [CanadaHonk/porffor#209]
I try to compile the richard.js sample under windows to native using: porf native richard.js richard.exe   --compiler=clang-cl 

The result is:  1513/1514  optimizing...TypeError: Cannot read properties of undefined (reading 'push')

**Comment by Rob23oba** [comment]
Could you try `-d`? That will hopefully give a better error message.

**Comment by Rob23oba** [comment]
Hmm, maybe try the option `--no-pgo`, that seems to be causing the issue.

**Comment by louisvangeldrop** [comment]
I can compile richards.js without any problem. However when I run richard.exe I do not get the result printed

**Comment by Rob23oba** [comment]
Yeah, there are still some issues to be fixed in the 2c compiler and in Porffor general.

**Comment by louisvangeldrop** [comment]
Will compileing to wasm help. And if so which wasm-engine to use?

**Comment by Rob23oba** [comment]
I'm pretty sure porffor just doesn't support some of the things that are used in there. You can try to compile to wasm but there are some details about importing that are a bit difficult, otherwise any wasm engine supporting exception handling should suffice. If you just want to run it, doing `porf richard.js` should do the trick but I'm going to assume that similar issues will arise.

**Comment by CanadaHonk** [comment]
All issues are being automatically closed as Porffor was recently almost entirely rewritten. Please reopen if still reproducable/relevant. See #344 for more information, thanks for understanding.
